src/libraries/Microsoft.PowerFx.Core/Public/Marshal/ITypeMashallerProvider.cs

```diff
@@ -0,0 +1,36 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using Microsoft.PowerFx.Core.Public.Types;
+using Microsoft.PowerFx.Core.Public.Values;
+
+namespace Microsoft.PowerFx.Core
+{
+    /// <summary>
+    /// Handles marshalling a given type. Invoked by the <see cref="TypeMarshallerCache"/>.
+    /// </summary>
+    public interface ITypeMashallerProvider
```

Comment:
>ITypeMashallerProvider [](http://example.com/codeflow?start=21&length=22)

spelling

Reply:
check file name spellings as well.

Reply:
What's the correct spelling?
I think it's: "Marshaller" / "Marshalling", "Marshal"

Reply:
That sounds right, yeah.